README.md

```diff
@@ -1,2 +1,50 @@
-# kvm-in-a-pod-demo
-Demo scripts / presentation
+# Kvm-in-a-pod-demo
+This demo showcases the use of a macvtap network backend for VMs using
+kubevirt.
+
+In this repo you'll find:
+  - [start cluster script](start_cluster.sh)
+  - [prepare nodes](prepare_node.sh)
+  - [manifests](two_macvtap_fedora_vms.yaml)
+
+We encourage prospective users to try out the demo, and use the provided
+[manifests](two_macvtap_fedora_vms.yaml) as reference for more advanced
+use cases.
+
+# Usage
+TODO
+
+# Installation
+TODO
+
+# How this works
+In this demo we spawn two fedora VMs using multus to plug an extra
+macvtap interface into the pod.
+
+The multus network-attachment-definition uses macvtap cni plus a resource
+exposed by the macvtap device plugin to provide a macvtap interface which will
+be used by the VMs.
+
+The overall flow happens as follows:
+  - KubeVirt takes the resource in the network attachment definition and places
+    it as a resource request in the pod definition.
+  - before the pod is spawned, kubelet requests the resource to the macvtap
+    device plugin.
+  - the macvtap device plugin allocates a new macvtap interface, and grants the
+    pod access to the corresponding /dev/tapX character device - via cgroups.
+  - KubeVirt generates the domxml libvirt will require to plug the character
+    device into the VM as a virtio interface.
```

Comment:
I would probably merge this point into the last, for sake of simplicity.

Reply:
Done.